modules/rpc/yarpc.go

```diff
@@ -177,13 +184,13 @@ func (c *dispatcherController) mergeConfigs() (conf yarpc.Config, err error) {
 		return conf, errors.New("unable to merge empty configs")
 	}
 
-	conf.Name = c.configs[0].AdvertiseName
```

Comment:
We want to discourage services using a different advertiseName but this removes the ability to do so. I think services might want this for eg. when a service is re-written, migration paths might involve advertising as a different service name.